src/Avalonia.MicroCom/MicroComVtblBase.cs

```diff
@@ -16,16 +16,23 @@ public unsafe class MicroComVtblBase
         public static IntPtr Vtable { get; } = new MicroComVtblBase().CreateVTable();
         public MicroComVtblBase()
         {
-            AddMethod((QueryInterfaceDelegate)QueryInterface);
-            AddMethod((AddRefDelegate)AddRef);
-            AddMethod((AddRefDelegate)Release);
+            AddMethod<QueryInterfaceDelegate>(QueryInterface);
```

Comment:
Those are explicitly marked with `[UnmanagedFunctionPointerAttribute(CallingConvention.StdCall)]`, no need to change anything there.

Reply:
The reason why I change this is to inform compiler that this specific type is used. Attribute `UnmanagedFunctionPointer` specify calling convention. That does not prevent AOT compiler from removing type information altogether. 

Should I need to change codegenerator then?

Reply:
I think I get what you are saying here and in the issue.

Reply:
>That does not prevent AOT compiler from removing type information altogether.

There is a cast to that delegate type, so type information is preserved since there is a direct reference from MSIL.

>Should I need to change codegenerator then?


The latest codegenerator uses `[UnmanagedCallersOnly]` for .NET 5+. We haven't enabled it yet because we can't switch to multitargeting in 0.10.x branch. For the next release we'll switch to [MicroCom.Runtime](https://www.nuget.org/packages/MicroCom.Runtime) package which is developed separately and [has those changes included](https://github.com/kekekeks/MicroCom/blob/35e7843626a41cd4d1b516b3768847ef642c71af/src/MicroCom.Runtime/MicroComVtblBase.cs#L20-L28)